server/src/main/java/org/opensearch/wlm/QueryGroupHelper.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.wlm;
+
+import org.opensearch.search.ResourceType;
+import org.opensearch.tasks.Task;
+
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Helper class for containing static utility methods related to QueryGroups.
+ */
+public class QueryGroupHelper {
+
+    /**
+     * A map that associates each {@link ResourceType} with a function that calculates the resource usage for a given {@link Task}.
+     */
+    private static final Map<ResourceType, Function<Task, Long>> resourceUsageCalculator = Map.of(
+        ResourceType.MEMORY,
+        (task) -> task.getTotalResourceStats().getMemoryInBytes(),
+        ResourceType.CPU,
+        (task) -> task.getTotalResourceStats().getCpuTimeInNanos()
+    );
```

Comment:
Having this kind of Map makes good case for keeping it in resourceType enum IMO. Not a particularly big fan of Helper classes. @kaushalmahi12 - I know you suggested this `QueryGroupHelper`, what do you think?

Reply:
@kaushalmahi12 pointed out that it would introduce unnecessary coupling if we add this in the ResourceType Enum and that makes sense.

Reply:
I am not convinced about the coupling part, since we are supplying these methods for each `ResourceType`, they might as well sit in the corresponding enum. Every `ResourceType` should be able to tell its usage from given `Task` object. This adds contraint that, for defining new `ResourceType` in future, we should also be able to tell how to compute its usage from `Task`. `QueryGroupHelper` does not ensure any such invariant!

Reply:
>This adds contraint that, for defining new ResourceType in future, we should also be able to tell how to compute its usage from Task

I think this is a very strong point, I agree that `QueryGroupHelper` does not guarantee such invariant. 
I am now convinced adding this Map in ResourceType. 

Making changes now. 
Thanks @jainankitk